tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -1002,6 +1025,184 @@ describe(__filename, () => {
+
+    expect(root.find('.UserProfileEdit-deletion-modal')).toHaveLength(0);
+
+    root.find('.UserProfileEdit-delete-button').simulate(
+      'click',
+      createFakeEvent()
+    );
+
+    expect(root.find('.UserProfileEdit-deletion-modal')).toHaveLength(1);
+
+    root.find('.UserProfileEdit-cancel-button').simulate(
+      'click',
+      createFakeEvent()
+    );
+
+    expect(root.find('.UserProfileEdit-deletion-modal')).toHaveLength(0);
+  });
+
+  it('dispatches deleteUserAccount and logOutUser when current logged-in user confirms account deletion', () => {
+    const clientApp = CLIENT_APP_FIREFOX;
+    const lang = 'fr-FR';
+
+    const userId = 456;
+    const username = 'tofumatt';
+    const params = { username };
+
+    const { store } = dispatchSignInActions({
+      clientApp,
+      lang,
+      userProps: {
+        ...defaultUserProps,
+        id: userId,
+        username,
+      },
+    });
+
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+
+    const root = renderUserProfileEdit({ errorHandler, params, store });
+
+    dispatchSpy.reset();
+
+    // User opens the modal.
+    root.find('.UserProfileEdit-delete-button').simulate(
+      'click',
+      createFakeEvent()
+    );
+
+    sinon.assert.notCalled(dispatchSpy);
+
+    // User confirms account deletion.
+    root.find('.UserProfileEdit-confirm-button').simulate(
+      'click',
+      createFakeEvent()
+    );
+
+    sinon.assert.callCount(dispatchSpy, 2);
+    sinon.assert.calledWith(dispatchSpy, deleteUserAccount({
+      errorHandlerId: errorHandler.id,
+      userId,
+    }));
+    sinon.assert.calledWith(dispatchSpy, logOutUser());
+
+    sinon.assert.calledWith(fakeRouter.push, `/${lang}/${clientApp}`);
+  });
+
+  it('does not dispatch logOutUser when current logged-in user confirms deletion of another user account', () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        ...defaultUserProps,
+        username: 'an-admin-user',
+        permissions: [USERS_EDIT],
+      },
+    });
+
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+
+    // Create a user with another username.
+    const user = createUserAccountResponse({ username: 'willdurand' });
+    store.dispatch(loadUserAccount({ user }));
+
+    const root = renderUserProfileEdit({
+      errorHandler,
+      params: { username: user.username },
+      store,
+    });
+
+    dispatchSpy.reset();
+
+    // User opens the modal.
+    root.find('.UserProfileEdit-delete-button').simulate(
+      'click',
+      createFakeEvent()
+    );
+
+    sinon.assert.notCalled(dispatchSpy);
+
+    // User confirms account deletion.
+    root.find('.UserProfileEdit-confirm-button').simulate(
+      'click',
+      createFakeEvent()
+    );
+
+    sinon.assert.callCount(dispatchSpy, 1);
+    sinon.assert.calledWith(dispatchSpy, deleteUserAccount({
```

Comment:
Just to be explicit, do you want to also do an assert.notCalled, or is that overkill?

Reply:
With both `callCount` and `calledWith`, I thought that it would be enough.